src/fetchers/tarball-fetcher.js

```diff
@@ -170,7 +172,17 @@ export default class TarballFetcher extends BaseFetcher {
           extractorStream,
         } = this.createExtractor(overwriteResolved, resolve, reject);
 
-        //
+        const handleRequestError = (res) => {
+          if (res.statusCode >= 400) {
+            // $FlowFixMe
+            const statusDescription = http.STATUS_CODES[res.statusCode];
+            reject(new Error(
+              reporter.lang('requestFailed', `${res.statusCode} ${statusDescription}`),
```

Comment:
It would be useful, but I don't think it's available in this callback. I'd have to get the body separately from the untar stream somehow.